bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java

```diff
@@ -381,6 +382,17 @@ public boolean start() throws IOException {
     @Override
     public List<SourceWithKeys> splitIntoBundles(
         long desiredBundleSizeBytes, PipelineOptions options) throws Exception {
+      // Update the desiredBundleSizeBytes in order to limit the
+      // number of splits to maximumNumberOfSplits.
+      long sizeEstimate = getEstimatedSizeBytes(options);
+      desiredBundleSizeBytes = Math.max(sizeEstimate / MAX_SPLIT_COUNT, desiredBundleSizeBytes);
```

Comment:
The hard limit is 20MB worth of serialized Source objects.  4,000 is a rough estimate.

Factoring in a 1/.75 inflation (33%), that leaves each serialized source to be a max of 3,750 bytes.  I did a quick test with a basic source.  It serialized to about 1K.  I had to work hard to get a complex enough FilterList that made it near 2K.

I think we're pretty safe at 4,000 as a really rough estimate.